src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/price/PriceUplifter.kt

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.price
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.pecs.jpc.config.TimeSource
+
+/**
+ * Domain level service to perform the actual price adjustments for a supplier.
+ *
+ * Prices for the supplied effective year are calculated based on prices for the previous year with the supplied multiplier.
+ */
+@Component
+class PriceUplifter(
+  private val priceRepository: PriceRepository,
+  private val priceUpliftRepository: SupplierPriceUpliftRepository,
+  private val timeSource: TimeSource
+) {
+
+  private val logger = LoggerFactory.getLogger(javaClass)
+
+  /**
+   * Any exception thrown calling this function will passed onto the onFailure lambda function.
+   */
+  internal fun uplift(
+    supplier: Supplier,
+    effectiveYear: Int,
+    multiplier: Double,
+    onFailure: (e: Throwable) -> Unit,
+    onSuccess: (upliftedPriceCount: Int) -> Unit
+  ) {
+    Result.runCatching {
```

Comment:
This is running in a non blocking Kotlin co-routine.  This process actually takes some time.  We cannot make the process quicker it takes time.  The UI will be designed in such a way as to make this clear.